lib/ui/platform_dispatcher.dart

```diff
@@ -1314,6 +1371,100 @@ class WindowPadding {
   }
 }
 
+/// Area of the display that may be obstructed by a hardware feature.This is
+/// populated only on Android.
+///
+/// The [bounds] are measured in logical pixels. On devices with two screens the
+/// coordinate system starts with [0,0] in the top-left corner of the left screen
+/// and expands to include the visual space between the screens and the right screen.
+/// The [type] can be used to determine if this display feature obstructs the screen or not.
+/// For example, [DisplayFeatureType.hinge] and [DisplayFeatureType.cutout] both obstruct the display,
+/// while [DisplayFeatureType.fold] does not.
+class DisplayFeature {
+  const DisplayFeature({
+    required this.bounds,
+    required this.type,
+    required this.state,
+  });
+
+  /// Area of the view occupied by this display feature, measured in logical pixels.
+  ///
+  /// On devices with two screens, the Flutter view spans from the top-right corner
+  /// of the left screen to the bottom-right corner of the right screen, including
+  /// the area occupied by any display feature. Bounds of display features are
+  /// reported in this coordinate system.
+  ///
+  /// For example, on a dual screen device in portrait mode:
+  ///
+  /// - [bounds.left] gives you the size of left screen, in logical pixels.
+  /// - [bounds.right] gives you the size of the left screen + the hinge width.
+  final Rect bounds;
+
+  /// Type of display feature, e.g. hinge, fold, cutout.
+  final DisplayFeatureType type;
+
+  /// Posture of display feature, which is populated only for folds and hinges.
+  /// For cutouts, this is [DisplayFeatureState.unknown]
+  final DisplayFeatureState state;
+
+  @override
+  bool operator ==(Object other) {
```

Comment:
Nit, this has typically been implemented as something like:
```
     if (identical(this, other))
        return true;
     if (other.runtimeType != runtimeType)
        return false;
     return other is DisplayFeature ...
```
Nothing wrong with your approach, but prefer consistency with rest of codebase